pypsa/optimization/constraints.py

```diff
@@ -62,19 +62,64 @@ def define_operational_constraints_for_non_extendables(
         dispatch_lower = (1, dispatch_lower), (-1, loss)
         dispatch_upper = (1, dispatch_upper), (1, loss)
     n.model.add_constraints(
-        dispatch_lower, ">=", lower, name=f"{c}-fix-{attr}-lower", mask=active
+        dispatch_lower, ">=", lower, name=f"{c}-fix-non-comm-{attr}-lower", mask=active
     )
     n.model.add_constraints(
-        dispatch_upper, "<=", upper, name=f"{c}-fix-{attr}-upper", mask=active
+        dispatch_upper, "<=", upper, name=f"{c}-fix-non-comm-{attr}-upper", mask=active
     )
 
 
-def define_operational_constraints_for_extendables(
+def define_operational_constraints_for_extendables_but_non_committables(
```

Comment:
Why do we need all these cases? Can't we have some as special cases of more general formulations?

Reply:
[Uploading PR_comment.docx.pdf…]()